src/v/serde/serde.h

```diff
@@ -665,6 +673,11 @@ void read_nested(iobuf_parser& in, T& t, std::size_t const bytes_left_limit) {
           "rounding issues.");
         auto rep = read_nested<int64_t>(in, bytes_left_limit);
         t = std::chrono::duration_cast<Type>(std::chrono::nanoseconds{rep});
+    } else if constexpr (is_chrono_time_point<Type>) {
+        static_assert(
+          always_false<Type>,
+          "Time point serialization is risky and can have unintended "
+          "consequences. Check with RedPanda team before fixing this.");
```

Comment:
one of these is also in the previous commit. perhaps you intended to move it into this commit?

also, why can we not do this at the top of the function:

```
static_assert(!is_chrono_time_point<Type>, "...");
```

it doesn't look like we need to assert false in the true condition.